src/index.ts

```
@@ -0,0 +1,181 @@
+const tronweb = require('tronweb');
+const tronproto = require('./protobuf/tron_pb');
+const contractproto = require('./protobuf/Contract_pb');
+
+import * as assert from 'assert';
+
+// this is a construct unique to tron
+export type ByteArray = number[];
+export type Transaction = { txID: string, signature?: string[] };
+
+export default class Tron {
+
+  constructor(coinName: string) {
+    // TODO: this will probably need to become more useful if we need to decode if based on network
+  }
+
+  static decodeContract(base64EncodedHex: string): DecodedContract | ParsedAccountUpdatePermission {
+    const rawContract = this.getRawTransferContract(base64EncodedHex);
+    // there should not be multiple contracts in this data
+    assert(rawContract.contractList.length === 1);
```

Please throw an Error with message, otherwise someone will have to debug this: `AssertionError [ERR_ASSERTION]: false == true`

In general, `assert()` should be avoided in non-test code, and even then the methods from shouldjs should be used.

---

ah shoot, good point, yeah these asserts should have never made it into live code